### ISSUE TITLE
[Feature] 不支持 bool Tensor 作为  kernel 入参

### ISSUE THREAD

**Issue by bearfly2** [tile-ai/tilelang-ascend#1270]
## Feature Description
在 TileLang GPU中，kernel 可以直接接收 `bool` 类型 Tensor 参数，但是在 TileLang Ascend 后端中，使用 bool tensor 参数会在 codegen 阶段失败。
报错位置：
tilelang-ascend/src/target/codegen_ascend.cc
getType(...)
Unsupported data type: bool

当前规避方式为在 Python wrapper 中把 torch.bool 转成 torch.int32，然后把 kernel 参数改成 int32
## Use Case

    @T.prim_func
    def main(mask: T.Tensor((n,), "bool")):
        with T.Kernel(n, is_npu=True) as (pid, _):
            if mask[pid]:
                pass

    return main

## Proposed Solution
如果 AscendC 不直接支持 bool 类型的 GlobalTensor，也可以在后端内部将 bool tensor 映射为 uint8_t / int8_t，并在条件判断中按非零值处理。

## Feature Type
- [ ] New operator/kernel (e.g., new GEMM variant, new attention mechanism)
- [ ] New API/primitive (e.g., new T.xxx operation)
- [ ] Compiler optimization (e.g., new pass, performance improvement)
- [ ] New backend support (e.g., new hardware target)
- [ ] Developer tooling (e.g., debugging, profiling)
- [ ] Documentation improvement
- [ ] Other: [specify]

## Operator Details (if applicable)

**Operator Type:**
[e.g. GEMM, Flash Attention, Softmax, LayerNorm, Convolution, etc.]

**Input/Output Specification:**
```
Input shapes and dtypes:
- A: [M, K], float16
- B: [K, N], float16

Output shapes and dtypes:
- C: [M, N], float16
```

**Performance Requirements:**
[e.g. target throughput, latency, memory bandwidth]

**Reference Implementation (if any):**
[e.g. PyTorch implementation, CUDA kernel, paper reference]

## Alternative Solutions
Describe any alternative solutions or features you've considered.

## Implementation Constraints (if known)

**Programming Mode Preference:**
- [ ] Developer mode (automatic)
- [ ] Expert mode (manual control)
- [ ] Both modes supported

**Memory Constraints:**
[e.g. L1 buffer size, UB buffer size constraints]

**Target Hardware:**
- [ ] Ascend A2
- [ ] Ascend A3
- [ ] Other Ascend variants

## Additional Context
Add any other context, references, or screenshots about the feature request here.

## Willingness to Contribute
- [ ] I'm willing to submit a PR for this feature
- [ ] I can help with testing
- [ ] I can provide reference implementations
- [ ] I need guidance to get started
